cmake/external.cmake

```diff
@@ -105,3 +105,95 @@ function(link_libjpeg target_name)
         )
     endif()
 endfunction()
+
+# libabsl
+# HINT: System abseil should be built with -DCMAKE_CXX_STANDARD=17
```

Comment:
This sounds like each app should have its own copy of abseil built with cmake standard they're using

Reply:
I'm, honestly not sure what to make of this. All I know is that a couple of major distributions [include it](https://repology.org/project/abseil-cpp/versions) including debian, fedora, nixos and gentoo.
Debian [builds it with `-DCMAKE_CXX_STANDARD=14`](https://salsa.debian.org/debian/abseil/-/blob/master/debian/rules)
Nixos [doesn't seem to add any additional options](https://github.com/NixOS/nixpkgs/blob/nixos-20.09/pkgs/development/libraries/abseil-cpp/default.nix#L17) (though I'm not sure if this flag would be passed implicitly anyway)
Fedora [doesn't seem to be doing anything either](https://src.fedoraproject.org/rpms/abseil-cpp/blob/rawhide/f/abseil-cpp.spec) (ditto)
A package on the AUR [builds it with `-DCMAKE_CXX_STANDARD=17`](https://aur.archlinux.org/cgit/aur.git/tree/PKGBUILD?h=abseil-cpp)
Gentoo [has a USE flag to build with `-DCMAKE_CXX_STANDARD=17`](https://gitweb.gentoo.org/repo/gentoo.git/tree/dev-cpp/abseil-cpp/abseil-cpp-20200923.2.ebuild#n60). Currently, this is only required by [mozc](https://gitweb.gentoo.org/repo/gentoo.git/tree/app-i18n/mozc/mozc-2.26.4220_p20201212102434_p20201219202429.ebuild#n56). Chromium and grpc build fine without the flag (but don't require it to be disabled!), and tensorflow has a local copy.

I've added this for completeness as it seems widely supported, and because upstream provides stable/reliable releases, even though depending on how the distro builds it it might or might not work. I'm not sure if this flag is the only way to solve the linking issues I got when linking telegram, nor if maybe -std=c++14 also works, but now I've got it working, leaving a note behind for whatever packager tries to debug this seems the most sensible to me. Upstream documentation doesn't seem to provide much information on whether mixing different C++ standards across the ABI boundary is ok or not.